.devcontainer/Dockerfile

```diff
@@ -333,40 +343,46 @@ RUN apt-key add /tmp/jfrog.pub && \
 
 # add patch that is missing in jfrog version of aioeventlet (it only comes with 1 of 2 patches)
 COPY lte/gateway/deploy/roles/magma/files/patches/aioeventlet_fd_exception.patch /tmp/
-RUN patch -N -s -f /usr/local/lib/python3.8/dist-packages/aioeventlet.py < /tmp/aioeventlet_fd_exception.patch && \
+RUN patch -N -s -f /usr/local/lib/python${PYTHON_VERSION}/dist-packages/aioeventlet.py < /tmp/aioeventlet_fd_exception.patch && \
     rm -rf /tmp/*
 
 ### install eggs (lte, orc8r)
 COPY /lte/gateway/python/ ${MAGMA_ROOT}/lte/gateway/python/
 WORKDIR ${MAGMA_ROOT}/lte/gateway/python/
-RUN pip3 install --quiet --upgrade --no-build-isolation --no-cache-dir --verbose --editable .[dev] && \
+RUN . ${PYTHON_VENV}/bin/activate && \
+    pip3 install --quiet --upgrade --no-build-isolation --no-cache-dir --verbose --editable .[dev] && \
     rm -rf lte.egg-info
 
 COPY /orc8r/gateway/python/ ${MAGMA_ROOT}/orc8r/gateway/python/
 WORKDIR ${MAGMA_ROOT}/orc8r/gateway/python/
-RUN pip3 install --quiet --upgrade --no-build-isolation --no-cache-dir --verbose --editable .[dev] && \
+RUN . ${PYTHON_VENV}/bin/activate && \
+    pip3 install --quiet --upgrade --no-build-isolation --no-cache-dir --verbose --editable .[dev] && \
     rm -rf orc8r.egg-info
 
-# install formatter autopep8
-RUN pip install --no-cache-dir autopep8
+### install formatter autopep8
+RUN . ${PYTHON_VENV}/bin/activate && \
+    pip3 install --no-cache-dir autopep8
+
+# TODO Does virtualenv have to be sourced in every RUN statement or could I just use $PYTHON_VENV/bin/pip3 instead? First option seems cleaner, the second one comes with less code.
```

Comment:
If the second is possible, this might be preferable in this case and is imho clear enough. I would guess this is easy to test locally.

Reply:
Tested it and it is possible. To be on the safe side I am now using `path/to/python -m pip` for which I introduced a variable `PYTHON_VENV_EXECTUABLE` after all ;)